### ISSUE TITLE
Multiple initiative rolls at the same time result in lost rolls

### ISSUE THREAD

**Issue by LtChae** [SkyJedi/SWEotE-Discord-Dice-Roller#27]
How to reproduce:
Make two initiative rolls in quick succession:
![proof](https://user-images.githubusercontent.com/5122881/84970731-46ce1180-b0e9-11ea-80dd-677791efedbe.PNG)
For logging purposes, "Today" is 6/17/2020

Recommendation is to build a patch for the initiative order, and then merge it with whatever the current order is at the end.

**Comment by SkyJedi** [comment]
cant do much about it with how emojis work and the character limits on
messages.  results are tagged for each person too....

**Comment by LtChae** [comment]
I don't think that would matter, would it? The issue seems to be the time it takes to complete the roll. It gets the init order at the start, performs all the steps needed, and then replaces the init already stored with the one it builds. With two running at the same time there's a high chance that one will overwrite the other.

**Comment by SkyJedi** [comment]
oh, no. that won't happen results are written before the messages are sent

**Comment by LtChae** [comment]
Doesn't the roll function send messages while it does its thing?

**Comment by SkyJedi** [comment]
no, well yes, but each has its own process. i dont think you could send commands fast enough to confuse the bot/db

**Comment by LtChae** [comment]
It's distressingly easy. Maybe something has changed about the data api?

**Comment by SkyJedi** [comment]
oh yeah, this quick again.  lemme sleep on it.  till then roll slower

**Comment by LtChae** [comment]
Appreciate you taking a look at it!

Looks like the other process is waited on. That would mean that it has to finish doing the roll messages before it updates the init: https://github.com/SkyJedi/FFGNDS-Discord-Dice-Roller/blob/926c0a290c2874a484ad45d3a715ef95e84cada6/modules/SW.GENESYS/initiative.js#L25

**Comment by SkyJedi** [comment]
Ahhhh, if is one of those "Do I need this code? No I do not need this code.
I have deleted the code. Oh shit I need that code!" Moments.

I think I know the fix, will test in the morning.

**Comment by SkyJedi** [comment]
its _better_but not bullet proof.  I need to completely overhaul initiative to really fix the problem.  so now just have players space out imitative rolls

**Comment by LtChae** [comment]
Cool, iterative fixes get us there eventually.
